kani-compiler/src/codegen_cprover_gotoc/reachability.rs

```diff
@@ -0,0 +1,586 @@
+                    // Thread local fall into this category.
+                    match self.tcx.const_eval_resolve(ParamEnv::reveal_all(), un_eval, None) {
+                        // The `monomorphize` call should have evaluated that constant already.
+                        Ok(const_val) => const_val,
+                        Err(ErrorHandled::TooGeneric) => span_bug!(
+                            self.body.source_info(location).span,
+                            "Unexpected polymorphic constant: {:?}",
+                            literal
+                        ),
+                        Err(error) => {
+                            warn!(?error, "Error already reported");
+                            return;
+                        }
+                    }
+                }
+                // Nothing to do
+                ConstKind::Param(..) | ConstKind::Infer(..) | ConstKind::Error(..) => return,
+
+                // Shouldn't happen
+                ConstKind::Placeholder(..) | ConstKind::Bound(..) => {
+                    unreachable!("Unexpected constant type {:?} ({:?})", ct, ct.kind())
+                }
+            },
+        };
+        self.collect_const_value(val);
+    }
+
+    fn visit_const(&mut self, constant: Const<'tcx>, location: Location) {
+        trace!(?constant, ?location, "visit_const");
+        self.super_const(constant);
+    }
+
+    /// Collect function calls.
+    fn visit_terminator(&mut self, terminator: &Terminator<'tcx>, location: Location) {
+        trace!(?terminator, ?location, "visit_terminator");
+
+        let tcx = self.tcx;
+        match terminator.kind {
+            TerminatorKind::Call { ref func, .. } => {
+                let callee_ty = func.ty(self.body, tcx);
+                let fn_ty = self.monomorphize(callee_ty);
+                if let TyKind::FnDef(def_id, substs) = *fn_ty.kind() {
+                    let instance =
+                        Instance::resolve(self.tcx, ParamEnv::reveal_all(), def_id, substs)
+                            .unwrap()
+                            .unwrap();
+                    self.collect_instance(instance, true);
+                } else {
+                    assert!(
+                        matches!(fn_ty.kind(), TyKind::FnPtr(..)),
+                        "Unexpected type: {:?}",
+                        fn_ty
+                    );
+                }
+            }
+            TerminatorKind::Drop { ref place, .. }
+            | TerminatorKind::DropAndReplace { ref place, .. } => {
+                let place_ty = place.ty(self.body, self.tcx).ty;
+                let place_mono_ty = self.monomorphize(place_ty);
+                let instance = Instance::resolve_drop_in_place(self.tcx, place_mono_ty);
+                self.collect_instance(instance, true);
+            }
+            TerminatorKind::InlineAsm { .. } => {
+                // We don't support inline assembly. Skip for now.
```

Comment:
Add a `todo!`?

Reply:
I don't want this to fail compilation though. The function we are visiting right now will still be collected for code generation and I assume that this instruction will get translated into an unsupported construct.

Reply:
I will improve the comment though.